### PR TITLE
Add comparison with Solargraph

### LINKED ISSUE #419: Comparison with Solargraph
I'm pretty new to language servers, and I've been hearing only about Solargraph so far. Since this seems to be a replacement, it would be useful if the README included some advantages over Solargraph, to help developers decide 🙂

### PR DESCRIPTION
**Pull request by andyw8**
Closes #419

My exposure to Solargraph is limited so some aspects here could be incorrect or incomplete.

## Motivation

Both Solargraph and ruby-lsp have similar goals, so it's not clear why a developer should pick one over the other. This is an attempt to give some useful information for an informed and objective decision.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,6 +93,71 @@ Possible values are:
 * `messages`: display requests and responses notifications
 * `verbose`: display each request and response as JSON
 
+## How does ruby-lsp differ from Solargraph?
```
[the remark this answers is not in the document]
Comment (shopify-aiden) [reply README.md:96]: I'm personally fine with the length of it, but could see benefit of adding a comparison page to leave room for future edits.

**Review by shopify-aiden — APPROVED** [review]
This is an awesome write up, thanks for giving some great context on the difference of the two. I think this would be really helpful for anyone assessing which they would want to use. 

Are there any other extensions in this space that we would want to add to this in the future? I think that might be a good indicator on if we want to add this to a different page or keep in int the README.

**Review by st0012 — COMMENTED** [review]
I like the idea 👍 
But is it possible to utilise tables here? Having stuff visually side-by-side via tables will make it even easier to compare them.

File: README.md [hunk 2]
```diff
@@ -93,6 +93,70 @@ Possible values are:
...
+
+**Background**
+
+In July 2021, Shopify quietly released [rubocop-lsp](https://github.com/Shopify/rubocop-lsp). In March 2022, that was succeeded by [ruby-lsp](https://github.com/Shopify/ruby-lsp).
```
Comment (vinistock) [inline README.md:100]: Is it worth mentioning the `rubocop-lsp` here? It was basically an experiment that is already archived.

File: README.md [hunk 3]
```diff
@@ -93,6 +93,70 @@ Possible values are:
...
+
+**Major Differences**
+
+The biggest difference between the two libraries is their approach to understanding the structure your code, and its dependencies.
+This is used to provide capabilities such as intellisense (context-aware auto-completion) and navigation (e.g. Go To Definition).
+
+Solargraph relies on [YARD](https://yardoc.org/), a popular tool for building documentation.
+Developers run `yard gems` command on their local machine to generate this information
+
+ruby-lsp does not provide direct support for these features.
```
Comment (vinistock) [inline README.md:117]: The Ruby LSP will provide those features eventually. We choose to use it with Sorbet for more accurate results, but it's not a requirement.

I would say something along the lines of `While the Ruby LSP does not yet implement these features, the goal is to have it index Ruby codebases to provide them without requiring any form of annotations`.

File: README.md [hunk 4]
```diff
@@ -93,6 +93,70 @@ Possible values are:
...
+The intention is that they are handled by Sorbet’s LSP server, running alongside ruby-lsp.
+This means some functionality is only available if the files are [typed at ‘true’ or higher](https://sorbet.org/docs/static#file-level-granularity-strictness-levels).
+However there is some experimental work [in progress](https://github.com/Shopify/ruby-lsp/pull/429) to improve support for untyped code.
```
Comment (vinistock) [inline README.md:120]: Sorbet is not required to use the Ruby LSP, even though you can use both at the same time. I wouldn't mention this.

File: README.md [hunk 5]
```diff
@@ -93,6 +93,70 @@ Possible values are:
...
+* Both libraries integrate with RuboCop for linting and auto-formatting. (Only ruby-lsp has a Quick Fix feature, allowing individual corrections.).
+* Both support Code Folding
+
+**Other differences**
+
+Features only in ruby-lsp:
+
+* Selection ranges (Expand Selection / Shrink Selection)
+* Inlay hints
+* Snippets
```
Comment (vinistock) [inline README.md:135]: If folks decide to implement these in Solargraph, we will need to maintain our own documentation in sync. I'm not a big fan of that...maybe we should focus on bigger, architectural and principle differences?

File: README.md [hunk 6]
```diff
@@ -93,6 +93,70 @@ Possible values are:
...
+
+Features only in Solargraph:
+
+* [Plugin support](https://solargraph.org/guides/plugins)
```
Comment (vinistock) [inline README.md:139]: This is planned for the Ruby LSP https://github.com/Shopify/ruby-lsp/issues/443.

**Review by vinistock — COMMENTED** [review]
I agree with Stan that a side by side table would be easier to digest

**Comment by andyw8** [comment]
Closing in favor of https://github.com/Shopify/ruby-lsp/pull/447

**Pull request closed without merging**